subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/wifi_mgmt_ext.h>
+#include <net/wifi_credentials.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+
+#include "includes.h"
+#include "common.h"
+#include "common/defs.h"
+#include "wpa_supplicant/config.h"
+#include "wpa_supplicant_i.h"
+#include "driver_i.h"
+
+extern struct k_sem wpa_supplicant_ready_sem;
+extern struct k_mutex wpa_supplicant_mutex;
+extern struct wpa_global *global;
+
+
+LOG_MODULE_REGISTER(wifi_mgmt_ext, CONFIG_WIFI_MGMT_EXT_LOG_LEVEL);
+
+/* helper functions {*/
+
+static inline struct wpa_supplicant *get_wpa_s_handle(const struct device *dev)
+{
+	struct wpa_supplicant *wpa_s = NULL;
+	int ret = k_sem_take(&wpa_supplicant_ready_sem, K_SECONDS(2));
+
+	if (ret) {
+		LOG_ERR("%s: WPA supplicant not ready: %d", __func__, ret);
+		return NULL;
+	}
+
+	k_sem_give(&wpa_supplicant_ready_sem);
+
+	wpa_s = wpa_supplicant_get_iface(global, dev->name);
+	if (!wpa_s) {
+		LOG_ERR("%s: Unable to get wpa_s handle for %s\n", __func__, dev->name);
+		return NULL;
+	}
+
+	return wpa_s;
+}
+
+static inline void band_to_freq(struct wifi_credentials_personal *creds, struct wpa_ssid *ssid)
```

Comment:
Is `inline` needed here and for other functions? Without constant arguments, the compiler is more likely to ignore the inline request

Reply:
removed